tools/sigclient/cmd/sigclient.go

```diff
@@ -146,6 +147,54 @@ var functionalTestCmd = &cobra.Command{
 	},
 }
 
+var performanceTestCmd = &cobra.Command{
+	Use:   "performance",
+	Short: "performance testing of SigLens",
+	Run: func(cmd *cobra.Command, args []string) {
+		dest, _ := cmd.Flags().GetString("dest")
+		queryDest, _ := cmd.Flags().GetString("queryDest")
+		bearerToken, _ := cmd.Flags().GetString("bearerToken")
+
+		log.Infof("dest : %+v\n", dest)
+		log.Infof("queryDest : %+v\n", queryDest)
+		log.Infof("bearerToken : %+v\n", bearerToken)
+
+		totalEvents := 1_000_000_000
+		batchSize := 50
+		numIndices := 10
+		processCount := 10
+		indexPrefix := "ind"
+		indexName := ""
+		numFixedCols := 100
+		maxVariableCols := 20
+		concurrentQueries := 5
+
+		logCh := make(chan utils.Log, 1000)
+
+		dataGenConfig := utils.InitPerformanceTestGeneratorDataConfig(numFixedCols, maxVariableCols, logCh)
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		go func(cancel context.CancelFunc) {
+			// addTs should be false for performance testing
+			ingest.StartIngestion(ingest.ESBulk, "performance", "", totalEvents, false, batchSize, dest, indexPrefix,
+				indexName, numIndices, processCount, false, 0, bearerToken, 0, 0, dataGenConfig)
+			cancel()
```

Comment:
When this is called here after ingestion, the query server is also stopped right? What if there are some logs in the channel and they need to be queried, but now they will not be queried because of calling `cancel` here?

Reply:
I believe we would all the logs that were sent in the channel, to be queried right?

Reply:
Logs channels are buffered for performance/(logs are ready to be queried) and the query server does not have to wait for it. If cancel is called, we essentially reach the end of the test. I think the remaining logs can be ignored since this is a long-running test.